### PR TITLE
fix(Accordion): Add separate component for Accordion print-view

### LINKED ISSUE #1380: Accordion looks weird in PDF view
### Description of the bug

When looking at an `Accordion` in PDF preview, the background of the text is not visible (or has very little height). 

### Steps To Reproduce

1. Go to frontend-tests.
2. Open PDF preview.
3. Observe the text inside accordion get these thin gray streaks above the content. 

### Additional Information

image

### PR DESCRIPTION
**Pull request by mikaelrss**
## Description
This PR fixes an issue with `Accordion` in PDF preview. 
It also adds the `headingLevel` options, which let's app developers specify the level of the heading element of the `Accordion`. 

## Related Issue(s)

- closes #1380 

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [x] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [x] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [x] Has been added/updated - [PR Here](https://github.com/Altinn/altinn-studio-docs/pull/1055)
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Changes/additions to component properties
  - [ ] Changes are reflected in both `src/layout/layout.d.ts` and `layout.schema.v1.json`, and these are all backwards-compatible
  - [x] No changes made
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Comment by mikaelrss** [comment]
I see that there are some Percy screenshots that are failing. Some are new screenshots, and some are old font. I cannot seem to see that any of the screenshots that are failing have any relevance to the changes proposed in this PR. 

I did however coincidentally see these gray boxes on some of the screenshots: 
image

My first thought was that this was a bug with the `SummaryAccordion` and `SummaryAccordionGroup` components, but I see that the screenshots are like this on the `main` branch as well. What do you think? Could this have a connection with `Accordion` or `AccordionGroup` somehow?

**Comment by bjosttveit** [comment]
> I see that there are some Percy screenshots that are failing. Some are new screenshots, and some are old font. I cannot seem to see that any of the screenshots that are failing have any relevance to the changes proposed in this PR.
> 
> I did however coincidentally see these gray boxes on some of the screenshots: image
> 
> My first thought was that this was a bug with the `SummaryAccordion` and `SummaryAccordionGroup` components, but I see that the screenshots are like this on the `main` branch as well. What do you think? Could this have a connection with `Accordion` or `AccordionGroup` somehow?

The grey boxes are areas we have defined that percy should ignore, for example the date and instance id changes depending on the run. So this has nothing to do with your PR. Also, percy is sometimes slow to update, so since the font was recently changed it has been lagging behind, and I have seen the exact same diff in other unrelated PRs.

TLDR; should be safe to ignore this!

File: src/layout/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -4,21 +4,26 @@ import { Accordion as DesignSystemAccordion } from '@digdir/design-system-react'
...
 import { AccordionItem } from 'src/layout/Accordion/AccordionItem';
 import { GenericComponent } from 'src/layout/GenericComponent';
 import type { PropsFromGenericComponent } from 'src/layout';
 
-type IAccordionProps = PropsFromGenericComponent<'Accordion'>;
+type IAccordionProps = PropsFromGenericComponent<'Accordion'> & { renderPDFPreview?: boolean };
 
-export const Accordion = ({ node }: IAccordionProps) => {
-  const { textResourceBindings, renderAsAccordionItem } = node.item;
+export const Accordion = ({ node, renderPDFPreview = true }: IAccordionProps) => {
```
Comment (olemartinorg) [inline src/layout/Accordion/Accordion.tsx:14]: The warning here says `renderPDFPreview` is not used anywhere - was this accidentally committed?

**Review by olemartinorg — APPROVED** [review]
Looks good to me! 🥳 Would like to get input from @bjosttveit on this as well, at least

File: src/layout/Accordion/Accordion.tsx (hunk 1 above)
Comment (mikaelrss) [reply src/layout/Accordion/Accordion.tsx:14]: Ah yes. This was an early idea. Will remove the unused prop.

File: src/features/pdf/data/generatePdfSagas.ts [hunk 2]
```diff
@@ -59,32 +59,28 @@ function generateAutomaticLayout(pdfFormat: IPdfFormat, uiConfig: IUiConfig, lay
...
-          id: `__pdf__${component.id}`,
+          id: `__pdf__${node.item.id}`,
           type: 'Summary',
-          componentRef: component.id,
-          pageRef,
+          componentRef: node.item.id,
+          pageRef: pageName,
           excludedChildren: pdfFormat?.excludedComponents,
-          largeGroup: component.type === 'Group' && (!component.maxCount || component.maxCount <= 1),
+          largeGroup: node.isNonRepGroup(),
```
[the remark this answers is not in the document]
Comment (bjosttveit) [reply src/features/pdf/data/generatePdfSagas.ts:83]: Good point, I totally missed that

[the remark this answers is not in the document]
Comment (bjosttveit) [reply src/features/pdf/data/generatePdfSagas.ts:83]: Turns out that `AccordionGroup` is currently defined as a `PresentationComponent`

[the remark this answers is not in the document]
Comment (mikaelrss) [reply src/features/pdf/data/generatePdfSagas.ts:83]: Ah. And it should probably be a `ContainerComponent`?

[the remark this answers is not in the document]
Comment (bjosttveit) [reply src/features/pdf/data/generatePdfSagas.ts]: @olemartinorg I changed it to be defined in the component classes instead (whether or not a component should be included in auto-pdf), Grid should be supported now.

@mikaelrss I guess that makes the most sense

**Review by bjosttveit — APPROVED** [review]
